test/test-browser-usage.js

```diff
@@ -0,0 +1,212 @@
+
+const TestServer = require('./helpers/test-server.js');
+
+require('geckodriver');
+
+require('chai').should();
+
+describe('Test Usage of Browsers', function() {
+  if ((process.env.TRAVIS || process.env.RELASE)) {
+    this.retries(3);
+  }
+
+  const DOWNLOAD_TIMEOUT = 5 * 60 * 1000;
+  const seleniumAssistant = require('../src/index.js');
+  const releases = ['stable', 'beta', 'unstable'];
+  const browserIds = ['chrome', 'firefox'];
+
+  if (!(process.env.TRAVIS || process.env.RELASE)) {
+    browserIds.push('opera');
+  }
+
+  // Travis has Safari, but the extension won't be installed :(
+  if ((!process.env.TRAVIS || process.env.RELASE) && process.platform === 'darwin') {
+    browserIds.push('safari');
+  }
+
+  let globalDriver = null;
+  let globalServer = new TestServer(false);
+  let localURL = '';
+
+  before(function() {
+    this.timeout(5 * 60 * 1000);
+
+    seleniumAssistant.setBrowserInstallDir(null);
+
+    return Promise.all([
+      seleniumAssistant.downloadBrowser('chrome', 'stable', 24),
+      seleniumAssistant.downloadBrowser('chrome', 'beta', 24),
+      seleniumAssistant.downloadBrowser('chrome', 'unstable', 24),
+      seleniumAssistant.downloadBrowser('firefox', 'stable', 24),
+      seleniumAssistant.downloadBrowser('firefox', 'beta', 24),
+      seleniumAssistant.downloadBrowser('firefox', 'unstable', 24)
+    ])
+    .catch(err => {
+      console.warn('There was an issue downloading the browsers: ', err);
+    })
+    .then(() => {
+      const serverPath = path.join(__dirname, 'data', 'example-site');
+      return globalServer.startServer(serverPath);
+    })
+    .then(portNumber => {
+      localURL = `http://localhost:${portNumber}/`;
+    });
+  });
+
+  beforeEach(function() {
+    // Timeout is to account for slow closing of selenium web driver browser
+    this.timeout(180000);
+
+    return seleniumAssistant.killWebDriver(globalDriver).catch(() => {})
+    .then(() => {
+      globalDriver = null;
+    });
+  });
+
+  after(function() {
+    this.timeout(6000);
+
+    return seleniumAssistant.killWebDriver(globalDriver).catch(() => {})
+    .then(() => {
+      return globalServer.killServer();
+    });
+  });
+
+  function isBlackListed(specificBrowser) {
+    // Return true to blacklist
+    return false;
+  }
+
+  function testNormalSeleniumUsage(specificBrowser) {
+    return specificBrowser.getSeleniumDriver()
+    .then(driver => {
+      globalDriver = driver;
+    })
+    .then(() => {
+      return globalDriver.get(localURL)
+      .then(() => {
+        return globalDriver.wait(selenium.until.titleIs('Example Site'), 1000);
+      });
+    })
+    .then(() => {
+      return seleniumAssistant.killWebDriver(globalDriver);
+    })
+    .then(() => {
+      globalDriver = null;
+    });
+  }
+
+  function testBuilderSeleniumUsage(specificBrowser) {
+    const builder = specificBrowser.getSeleniumDriverBuilder();
+
+    return builder.buildAsync()
+    .then(driver => {
```

Comment:
I'm not going to explicitly call out each instance, but in general, I'd recommend moving to single line arrow functions, taking advantage of the implicit `return` behavior, for succinctness and overall readability whenever possible.